logstash-core/lib/logstash/api/modules/node.rb

```diff
@@ -56,7 +56,6 @@ def node
           opts = {:graph => as_boolean(params.fetch("graph", false)),
                   :vertices => as_boolean(params.fetch("vertices", false))}
           payload = node.pipelines(opts)
-          halt(404) if payload.empty?
```

Comment:
When getting a resource that is a list, and that list is empty, we should return the empty list in a success response.